File: py/2482.py
class Solution:
    def onesMinusZeros(self, grid: List[List[int]]) -> List[List[int]]:
        onesRow = [0 for _ in range(len(grid))]
        zerosRow = [0 for _ in range(len(grid))]
        onesCol = [0 for _ in range(len(grid[0]))]
        zerosCol = [0 for _ in range(len(grid[0]))]

        for i in range(len(grid)):
            for k in range(len(grid[0])):
                if grid[i][k] == 0:
                    zerosRow[i] += 1
                    zerosCol[k] += 1
                else:
                    onesRow[i] += 1
                    onesCol[k] += 1

        diff = [[] for _ in range(len(grid))]
        for i in range(len(grid)):
            for k in range(len(grid[0])):
                diff[i].append(onesRow[i] + onesCol[k] - zerosRow[i] - zerosCol[k])
        return diff

File: cpp/2482.cpp
class Solution {

  public:
    vector<vector<int>> onesMinusZeros(vector<vector<int>>& grid) {
      vector<int> onesRow(grid.size(), 0);
      vector<int> onesCol(grid[0].size(), 0);
      vector<int> zerosRow(grid.size(), 0);
      vector<int> zerosCol(grid[0].size(), 0);

      for (int i = 0; i < grid.size(); i++) {
        for (int j = 0; j < grid[0].size(); j++) {
          if (grid[i][j] == 1) {
            onesRow[i]++;
            onesCol[j]++;
          }
          else {
            zerosRow[i]++;
            zerosCol[j]++;
          }
        }
      }

      vector<vector<int>> diff(grid.size());
      for (int i = 0; i < grid.size(); i++)
        for (int j = 0; j < grid[0].size(); j++)
          diff[i].push_back(onesRow[i] + onesCol[j] - zerosRow[i] - zerosCol[j]);
      return diff;
    }

};
